### PR TITLE
Us1.3 incl. integration and service class tests incl. basic frontend is connected

### PR DESCRIPTION
**Pull request by ThiemoOrrico**

### CODE REVIEW

File: backend/src/main/java/com/tradomator/model/CoinIdCard.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+import java.time.LocalDateTime;
+
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+//@Builder
```
Comment (Cobas91) [inline backend/src/main/java/com/tradomator/model/CoinIdCard.java:12]:
```suggestion

```

File: backend/src/main/java/com/tradomator/service/CoinService.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        this.coingeckoMapper = coingeckoMapper;
+    }
+
+    public CoinIdCard getCoinDataById(String coinId) {
+        CgeckoApiCoin cgeckoApiCoins = coinGeckoApiService.getCgeckoCoinById(coinId);
+        return coingeckoMapper.mapToCoin(cgeckoApiCoins);
+    }
+
+    public List<CoinIdCard> getAllCoinData() {
+        CgeckoApiCoin[] cgeckoApiCoins = coinGeckoApiService.getAllCgeckoCoinInfo();
```
Comment (Cobas91) [inline backend/src/main/java/com/tradomator/service/CoinService.java]:
```suggestion
        List<CgeckoApiCoin> cgeckoApiCoins = coinGeckoApiService.getAllCgeckoCoinInfo();
```
I would map the API Response into a List to work with in your Service Layer.

File: frontend/src/App.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import * as React from 'react';
+import LayoutGrid from "./components/LayoutGrid";
+
+
+
+export default function App() {
+    return (
+        <div className="App">
+    <LayoutGrid/>
```
Comment (Cobas91) [inline frontend/src/App.js]:
```suggestion
    <LayoutGrid/>
```
This name is a little bit confusing on the upper layer

File: frontend/src/App.test.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import { render, screen } from '@testing-library/react';
+import App from './App';
+
+test('renders learn react link', () => {
+  render(<App />);
+  const linkElement = screen.getByText(/learn react/i);
+  expect(linkElement).toBeInTheDocument();
+
+
+});
```
Comment (Cobas91) [inline frontend/src/App.test.js]: I would delete this files you dont need

File: frontend/src/components/CryptoCard.js [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+            {coinData && (
+                <p>
+                    Last updated: {convertToReadableDate(coinData?.lastUpdated)}
+                </p>
+            )}
+        </Wrapper>
+    )
+}
+
```
Comment (Cobas91) [inline frontend/src/components/CryptoCard.js]: In a few frontend files you are using styled components in other you use the styling of the mui components. Try to get one level for styling

File: frontend/src/pages/Loginpage.js [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+    const handleChange = event => {
+        setCredentials({...credentials, [event.target.name]: event.target.value})
+    }
+
+    const handleSubmit = event => {
+        event.preventDefault()
+        login(credentials)
+    }
+
+    const clientId = "20bd6360403e3e1711a8"
```
Comment (Cobas91) [inline frontend/src/pages/Loginpage.js]:
```suggestion

```
Get your ClientIdfrom your backend

File: frontend/src/index.css [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+  -moz-osx-font-smoothing: grayscale;
+
+
+
+}
+
+code {
+  font-family: source-code-pro, Menlo, Monaco, Consolas, 'Courier New',
+    monospace;
+}
```
Comment (Cobas91) [inline frontend/src/index.css:16]: Could collide with your Global Styled you set. Delete this file or get the global styles in here

File: frontend/src/components/TempCardOverview.js [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+
+    return (
+
+
+          <div>
+
+                <CoingeckoOverview coins={coins} />
+
+          </div>
+
```
Comment (Cobas91) [inline frontend/src/components/TempCardOverview.js]: Get prettier for your code :-)

**Review by Cobas91 — APPROVED** [review]
Much of code but some genius patterns :-) Move on!

File: frontend/src/components/BottomNavigationToDelete.js [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+
+    return Array.from(new Array(50)).map(
+        () => messageExamples[getRandomInt(messageExamples.length)],
+    );
+}
+
+export default function FixedBottomNavigation() {
+    const [value, setValue] = React.useState(0);
+    const ref = React.useRef(null);
+    const [messages, setMessages] = React.useState(() => refreshMessages());
```
Comment (Flooooooooooorian) [inline frontend/src/components/BottomNavigationToDelete.js]: You dont need the React Keyword
```suggestion
    const [value, setValue] = useState(0);
    const ref = useRef(null);
    const [messages, setMessages] = useState(() => refreshMessages());
```

File: frontend/src/components/BottomNavigationToDelete.js [hunk 10]
```diff
@@ -0,0 +1,104 @@
+import * as React from 'react';
```
Comment (Flooooooooooorian) [inline frontend/src/components/BottomNavigationToDelete.js]:
```suggestion
import {useState, ....} from 'react';
```

File: backend/src/test/java/com/tradomator/service/CoinServiceTest.java [hunk 11]
```diff
@@ -0,0 +1,137 @@
...
+        coinIdCard1.setLastUpdated(currentDateTime);
+
+        when(coingeckoMapper.mapToCoin(cgeckoApiCoin1)).thenReturn(coinIdCard1);
+
+        //WHEN
+        CoinIdCard actual = coinService.getCoinDataById(coinId);
+
+        //THEN
+        assertThat(actual, is(coinIdCard1));
+
```
Comment (Flooooooooooorian) [inline backend/src/test/java/com/tradomator/service/CoinServiceTest.java]: verify your mock calls

File: backend/src/test/java/com/tradomator/service/CoinServiceTest.java [hunk 12]
```diff
@@ -0,0 +1,137 @@
...
+        when(coingeckoMapper.mapToCoins(new CgeckoApiCoin[] {cgeckoApiCoin1, cgeckoApiCoin2})).thenReturn(
+                List.of(coinIdCard1, coinIdCard2));
+
+        //WHEN
+        List<CoinIdCard> actual = coinService.getAllCoinData();
+
+        //THEN
+        assertThat(actual, is(List.of(coinIdCard1, coinIdCard2)));
+
```
Comment (Flooooooooooorian) [inline backend/src/test/java/com/tradomator/service/CoinServiceTest.java:132]: verify your mock calls

File: frontend/src/service/coingecko-service.js [hunk 13]
```diff
@@ -0,0 +1,9 @@
+import axios from 'axios'
+
+export const getMain = () => {
+    return axios.get('/coingecko-api').then(res => res.data)
+}
+
+export const getById = coinId => {
+    return axios.get(`coingecko-api/${coinId}`).then(res => res.data)
```
Comment (Flooooooooooorian) [inline frontend/src/service/coingecko-service.js]:
```suggestion
    return axios.get(`/coingecko-api/${coinId}`).then(res => res.data)
```

**Review by Flooooooooooorian — APPROVED** [review]

File: frontend/src/App.test.js (hunk 4 above)
Comment (ThiemoOrrico) [reply frontend/src/App.test.js]: Hi Stephan thanks for pointing out will delete

File: frontend/src/index.css (hunk 7 above)
Comment (ThiemoOrrico) [reply frontend/src/index.css:16]: Styling needs to be cleaned up still, try to focus on mui styling

File: frontend/src/components/TempCardOverview.js (hunk 8 above)
Comment (ThiemoOrrico) [reply frontend/src/components/TempCardOverview.js]: good point :-) on the list

File: backend/src/main/java/com/tradomator/service/CoinService.java (hunk 2 above)
Comment (ThiemoOrrico) [reply backend/src/main/java/com/tradomator/service/CoinService.java]: thanks good suggestion - future proof ;-)

File: backend/src/test/java/com/tradomator/service/CoinServiceTest.java (hunk 12 above)
Comment (ThiemoOrrico) [reply backend/src/test/java/com/tradomator/service/CoinServiceTest.java:132]: done thanks for the hint

**Pull request merged**